components/DataTable.tsx

```diff
@@ -37,7 +41,7 @@ export function DataTable<TData, TValue>({
   });
 
   return (
-    <Table {...tableProps}>
+    <Table {...tableProps} ref={tableRef}>
```

Comment:
I'm passing that `ref` as `tableRef` rather than using `forwardRef` because:
1. We may want to add more refs in the future (i.e. on rows, headers, etc)
2. I discovered `forwardRef` does not play well with generic functions/components out of the box: https://fettblog.eu/typescript-react-generic-forward-refs/. Doing this greatly simplifies the code.